opentelemetry-otlp: Please upgrade grpcio-sys to 0.8.1 for compilation issue with openssl-vendored

Attempting to build opentelemetry-otlp with features `"grpc-sys"` and `"openssl-vendored"` set produces a compilation error:

<details>

```text
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc: In function ‘tsi_result ssl_ctx_use_engine_private_key(SSL_CTX*, const char*, size_t)’:
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:613:5: error: ‘ENGINE_load_dynamic’ was not declared in this scope
    613 |     ENGINE_load_dynamic();
        |     ^~~~~~~~~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:614:14: error: ‘ENGINE_by_id’ was not declared in this scope
    614 |     engine = ENGINE_by_id(engine_name);
        |              ^~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:624:12: error: ‘ENGINE_ctrl_cmd_string’ was not declared in this scope
    624 |       if (!ENGINE_ctrl_cmd_string(engine, "ID", engine_name, 0) ||
        |            ^~~~~~~~~~~~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:634:37: error: ‘ENGINE_METHOD_ALL’ was not declared in this scope
    634 |     if (!ENGINE_set_default(engine, ENGINE_METHOD_ALL)) {
        |                                     ^~~~~~~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:634:10: error: ‘ENGINE_set_default’ was not declared in this scope
    634 |     if (!ENGINE_set_default(engine, ENGINE_METHOD_ALL)) {
        |          ^~~~~~~~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:639:10: error: ‘ENGINE_init’ was not declared in this scope
    639 |     if (!ENGINE_init(engine)) {
        |          ^~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:644:19: error: ‘ENGINE_load_private_key’ was not declared in this scope
    644 |     private_key = ENGINE_load_private_key(engine, key_id, 0, 0);
        |                   ^~~~~~~~~~~~~~~~~~~~~~~
  /home/josh/.cargo/registry/src/github.com-1ecc6299db9ec823/grpcio-sys-0.7.2/grpc/src/core/tsi/ssl_transport_security.cc:656:26: error: ‘ENGINE_free’ was not declared in this scope
    656 |   if (engine != nullptr) ENGINE_free(engine);
        |                          ^~~~~~~~~~~
  gmake[3]: *** [CMakeFiles/grpc.dir/build.make:5737: CMakeFiles/grpc.dir/src/core/tsi/ssl_transport_security.cc.o] Error 1
  gmake[2]: *** [CMakeFiles/Makefile2:992: CMakeFiles/grpc.dir/all] Error 2
  gmake[1]: *** [CMakeFiles/Makefile2:999: CMakeFiles/grpc.dir/rule] Error 2
```

</details>

grpcio-sys 0.8.1 upgrades to a version of the grpc library that should fix this problem. Would you consider updating opentelemetry-otlp to use grpcio-sys 0.8.1?

--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@ The Rust [OpenTelemetry](https://opentelemetry.io/) implementation.
 [![Documentation](https://docs.rs/opentelemetry/badge.svg)](https://docs.rs/opentelemetry)
 [![LICENSE](https://img.shields.io/crates/l/opentelemetry)](./LICENSE)
 [![GitHub Actions CI](https://github.com/open-telemetry/opentelemetry-rust/workflows/CI/badge.svg)](https://github.com/open-telemetry/opentelemetry-rust/actions?query=workflow%3ACI+branch%3Amain)
-[![codecov](https://codecov.io/gh/TommyCpp/opentelemetry-rust/branch/master/graph/badge.svg?token=5OYQK2KY8Z)](https://codecov.io/gh/TommyCpp/opentelemetry-rust)
+[![codecov](https://codecov.io/gh/open-telemetry/opentelemetry-rust/branch/main/graph/badge.svg)](https://codecov.io/gh/open-telemetry/opentelemetry-rust)
 [![Gitter chat](https://img.shields.io/badge/gitter-join%20chat%20%E2%86%92-brightgreen.svg)](https://gitter.im/open-telemetry/opentelemetry-rust)
 
 [Website](https://opentelemetry.io/) |

--- a/README.md
+++ b/README.md
@@ -10,6 +10,7 @@ The Rust [OpenTelemetry](https://opentelemetry.io/) implementation.
 [![Documentation](https://docs.rs/opentelemetry/badge.svg)](https://docs.rs/opentelemetry)
 [![LICENSE](https://img.shields.io/crates/l/opentelemetry)](./LICENSE)
 [![GitHub Actions CI](https://github.com/open-telemetry/opentelemetry-rust/workflows/CI/badge.svg)](https://github.com/open-telemetry/opentelemetry-rust/actions?query=workflow%3ACI+branch%3Amain)
+[![codecov](https://codecov.io/gh/TommyCpp/opentelemetry-rust/branch/master/graph/badge.svg?token=5OYQK2KY8Z)](https://codecov.io/gh/TommyCpp/opentelemetry-rust)
 [![Gitter chat](https://img.shields.io/badge/gitter-join%20chat%20%E2%86%92-brightgreen.svg)](https://gitter.im/open-telemetry/opentelemetry-rust)
 
 [Website](https://opentelemetry.io/) |

--- a/opentelemetry-otlp/Cargo.toml
+++ b/opentelemetry-otlp/Cargo.toml
@@ -37,7 +37,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 [dependencies]
 async-trait = "0.1"
 futures = "0.3"
-grpcio = { version = "0.7", optional = true }
+grpcio = { version = "0.8", optional = true }
 opentelemetry = { version = "0.14", default-features = false, features = ["trace"], path = "../opentelemetry" }
 prost = { version = "0.7", optional = true }
 protobuf = { version = "2.18", optional = true }

